src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -212,16 +269,19 @@ private bool TryGetRowIdWorker(SqlConnection connection, TDatabaseId dataId, out
             }
 
             private void InsertOrReplaceBlob(
-                SqlConnection conection, TDatabaseId dataId, byte[] bytes, int length)
+                SqlConnection conection, TDatabaseId dataId, 
+                byte[] checksumBytes, int checksumLength,
+                byte[] dataBytes, int dataLength)
             {
                 using (var resettableStatement = conection.GetResettableStatement(
```

Comment:
doesn't this prepared statement can be cahced? and only parameter get bound each time?

Reply:
yes.  that's how it works.  That's what  resettableStatement is.

Reply:
I see so you recreate query string everytime but actual statement is cached. but it is cached in dictionary which seems unbound and not protected by lock. what make it safe guard on concurrent usage?

Reply:
> I see so you recreate query string everytime but actual statement is cached

Yes.  We could def cache the string as well... and we probably should.  I'll make that change.

> but it is cached in dictionary which seems unbound and not protected by lock. what make it safe guard on concurrent usage?

A connection object can only be used by a single thread at a time.  That's the definition of a connectino.  It's how one thread accesses and uses the database.  It is not shareable.

Reply:
Ok.  Have moved to caching the string once so we don't have the string allocations either.  Thanks!